src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -0,0 +1,492 @@
+	private GridPane pane;
+
+	private Integer rowNo;
+
+	private DateTimeFormatter dateFmt;
+	private DateTimeFormatter dateTimeFmt;
+
+	private final double ID_LABEL_WIDTH = 50.0;
+	private final double NAME_LABEL_WIDTH = 350.0;
+	private final double DATE_LABEL_WIDTH = 200.0;
+	private final int NO_OF_COLUMNS = 3;
+
+	public TaskEventRowFactory(DataLists masterList, DataLists displayList, GridPane pane){
+		this.masterList = masterList;
+		this.displayList = displayList;
+		this.pane = pane;
+		this.rowNo = new Integer(0);
+
+		dateFmt = DateTimeFormatter.ofPattern("dd MMM yyyy");
+		dateTimeFmt = DateTimeFormatter.ofPattern("dd MMM h:mm a");
+	}
+
+	public void showTodayTaskEvents(){
+		rowNo = new Integer(-1);
+		pane.getChildren().clear();
+
+		Label headerLabel = new Label("Today");
+		headerLabel.getStyleClass().add("header");
+		pane.add(headerLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		int noOfTaskEvents = showTaskEventsOnDate(LocalDate.now());
+
+		Label tempEmptyLabel = new Label("");
+		pane.add(tempEmptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		if (noOfTaskEvents == 0){
+			pane.getChildren().remove(tempEmptyLabel);
+
+			Label noTaskEventsLabel = new Label("No Tasks or Events today");
+			noTaskEventsLabel.getStyleClass().add("no-overdue-label");
+			pane.add(noTaskEventsLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+		}
+	}
+
+	public void showTaskEventsFromDate(LocalDate selectedDate){
+		pane.getChildren().clear();
+		rowNo = new Integer(-1);
+
+		LocalDate tempDate = selectedDate;
+
+		while(!tempDate.isAfter(getLastDate())){
+			addTaskEventsOnDate(tempDate, selectedDate);
+			tempDate = tempDate.plusDays(1);
+		}
+
+		if(rowNo == -1){
+			addNoEventTaskLabel(selectedDate);
+		}
+
+	}
+
+	public int showTaskEventsOnDate(LocalDate date){
+		int noOfEvents = addEventOnDate(date);
+		int noOfTasks = addDeadlineTaskOnDate(date);
+
+		rowNo += (noOfEvents+noOfTasks);
+
+		Label emptyLabel = new Label("");
+		pane.add(emptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		if ((noOfEvents+noOfTasks) == 0){
+			pane.getChildren().remove(emptyLabel);
+		}
+
+		return noOfEvents + noOfTasks;
+	}
+
+	public int showOverdue(){
+		int noOfOverdue = 0;
+
+		for(Event e: masterList.getEventsList()){
+			if(checkOverdue(e)){
+				add(e);
+				noOfOverdue++;
+			}
+		}
+
+		for(DeadlineTask t: masterList.getDeadlineTasksList()){
+			if(checkOverdue(t)){
+				add(t);
+				noOfOverdue++;
+			}
+		}
+
+		if(noOfOverdue == 0){
+			Label noOverdueLabel = new Label("No Overdue Tasks or Events");
+			noOverdueLabel.getStyleClass().add("no-overdue-label");
+			pane.add(noOverdueLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+		} else {
+			rowNo += noOfOverdue;
+		}
+
+		return noOfOverdue;
+	}
+
+	public boolean showFloatingTasks(boolean showCompleted){
+		boolean hasCompleted = false;
+
+		for(FloatingTask t: masterList.getFloatingTasksList()){
+			if(!t.getCompleted()){
+				rowNo++;
+				add(t);
+			}
+			else {
+				hasCompleted = true;
+			}
+		}
+
+		if(showCompleted){
```

Comment:
`willShowCompleted` or `isShowCompleted` makes it more explicit that this is a boolean